src/Lib/Managers/NobManager.vala

```diff
@@ -0,0 +1,345 @@
+        }
+        nobs[Nob.TOP_RIGHT].raise (select_effect);
+
+        if (print_middle_height_nobs) {
+          // RIGHT CENTER nob
+          nobs[Nob.RIGHT_CENTER].set_transform (transform);
+          if (print_middle_width_nobs) {
+            nobs[Nob.RIGHT_CENTER].translate (x + width - (nob_offset - line_width), y + (height / 2) - nob_offset);
+          } else {
+            nobs[Nob.RIGHT_CENTER].translate (x + width + line_width, y + (height / 2) - nob_offset);
+          }
+          nobs[Nob.RIGHT_CENTER].set ("visibility", Goo.CanvasItemVisibility.VISIBLE);
+        } else {
+          nobs[Nob.RIGHT_CENTER].set ("visibility", Goo.CanvasItemVisibility.HIDDEN);
+        }
+        nobs[Nob.RIGHT_CENTER].raise (select_effect);
+
+        // BOTTOM RIGHT nob
+        nobs[Nob.BOTTOM_RIGHT].set_transform (transform);
+        if (print_middle_width_nobs && print_middle_height_nobs) {
+          nobs[Nob.BOTTOM_RIGHT].translate (x + width - (nob_offset - line_width), y + height - (nob_offset - line_width));
+        } else {
+          nobs[Nob.BOTTOM_RIGHT].translate (x + width + line_width, y + height + line_width);
+        }
+        nobs[Nob.BOTTOM_RIGHT].raise (select_effect);
+
+        if (print_middle_width_nobs) {
+          // BOTTOM CENTER nob
+          nobs[Nob.BOTTOM_CENTER].set_transform (transform);
+          if (print_middle_height_nobs) {
+            nobs[Nob.BOTTOM_CENTER].translate (x + (width / 2) - nob_offset, y + height - (nob_offset - line_width));
+          } else {
+            nobs[Nob.BOTTOM_CENTER].translate (x + (width / 2) - nob_offset, y + height + line_width);
+          }
+          nobs[Nob.BOTTOM_CENTER].set ("visibility", Goo.CanvasItemVisibility.VISIBLE);
+        } else {
+          nobs[Nob.BOTTOM_CENTER].set ("visibility", Goo.CanvasItemVisibility.HIDDEN);
+        }
+        nobs[Nob.BOTTOM_CENTER].raise (select_effect);
+
+        // BOTTOM LEFT nob
+        nobs[Nob.BOTTOM_LEFT].set_transform (transform);
+        if (print_middle_width_nobs && print_middle_height_nobs) {
+          nobs[Nob.BOTTOM_LEFT].translate (x - (nob_offset + line_width), y + height - (nob_offset - line_width));
+        } else {
+          nobs[Nob.BOTTOM_LEFT].translate (x - (nob_size + line_width), y + height + line_width);
+        }
+        nobs[Nob.BOTTOM_LEFT].raise (select_effect);
+
+        if (print_middle_height_nobs) {
+          // LEFT CENTER nob
+          nobs[Nob.LEFT_CENTER].set_transform (transform);
+          if (print_middle_width_nobs) {
+            nobs[Nob.LEFT_CENTER].translate (x - (nob_offset + line_width), y + (height / 2) - nob_offset);
+          } else {
+            nobs[Nob.LEFT_CENTER].translate (x - (nob_size + line_width), y + (height / 2) - nob_offset);
+          }
+          nobs[Nob.LEFT_CENTER].set ("visibility", Goo.CanvasItemVisibility.VISIBLE);
+        } else {
+          nobs[Nob.LEFT_CENTER].set ("visibility", Goo.CanvasItemVisibility.HIDDEN);
+        }
```

Comment:
Let's create a method to set the visibility so we don't have to repeat this and it's easier to update if things change.